### PR TITLE
BREAKING(testing/bench): deprecate testing/bench module

### LINKED ISSUE #2103: Deprecate and remove "testing/bench.ts" module
Not that `deno bench` subcommand is available I think we should deprecate and eventually remove `testing/bench.ts` module.

While the module is certainly useful I strongly believe we should encourage use of `Deno.bench()` API and add functionality to it if some is missing.

Let's deprecate it in Deno v1.21 release and remove in Deno v1.22

CC @littleof

### PR DESCRIPTION
**Pull request by kt3k**
closes #2103

### CODE REVIEW

**Comment by kt3k** [comment]
Needs #2106 to pass CI

File: testing/bench.ts [hunk 1]
```diff
@@ -147,7 +147,11 @@ function createBenchmarkTimer(clock: BenchmarkClock): BenchmarkTimer {
 
 const candidates: BenchmarkDefinition[] = [];
 
-/** Registers a benchmark as a candidate for the runBenchmarks executor. */
+/**
+ * @deprecated Use Deno.bench() instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```
Comment (bartlomieju) [inline testing/bench.ts:151]:
```suggestion
 * @deprecated Use `Deno.bench()` instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```

File: testing/bench.ts [hunk 2]
```diff
@@ -165,7 +169,11 @@ export function bench(
   }
 }
 
-/** Clears benchmark candidates which name matches `only` and doesn't match `skip`.
+/**
+ * @deprecated Use Deno.bench() instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```
Comment (bartlomieju) [inline testing/bench.ts:173]:
```suggestion
 * @deprecated Use `Deno.bench()` instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```

File: testing/bench.ts [hunk 3]
```diff
@@ -179,6 +187,9 @@ export function clearBenchmarks({
 }
 
 /**
+ * @deprecated Use Deno.bench() instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```
Comment (bartlomieju) [inline testing/bench.ts:190]:
```suggestion
 * @deprecated Use `Deno.bench()` instead. See https://doc.deno.land/deno/unstable/~/Deno.bench
```

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**